pkg/runtest/run_test.go

```diff
@@ -18,7 +18,7 @@ import (
 
 func Test(t *testing.T) {
 	switch runtime.GOOS {
-	case "openbsd":
+	case "freebsd", "openbsd":
```

Comment:
With few tweaks I was able to get it working:
https://gist.githubusercontent.com/dvyukov/6415281cb806160698609e98633a1565/raw/39477969b5a245dab751d40c9af4bd161317bab9/gistfile1.txt
What do you think? These tests give quite useful coverage of end-to-end test program execution. Otherwise it only happens during fuzzing and nobody knows if it really works or not.

Reply:
I dropped this commit from the PR.

Your patch seems to work for me, but I'm confused by the change to run.go - what exactly prompted it?

I suspect the change to add -B/usr/lib32 is a bug in our gcc port; it shouldn't be necessary when using the base system's copy of clang. Seems fine for the time being though.

Reply:
> Your patch seems to work for me, but I'm confused by the change to run.go - what exactly prompted it?

When executor executes syscalls and has a heuristic to detect when syscalls block. If a syscall blocks, then just starts the next syscall on another thread. This is to avoid programs deadlocking on a first blocking syscall (e.g. any empty pipe read) and to be able to actually test these blocking paths (e.g. later write to the pipe).
This heuristic is timeout based and is inherently flaky:
https://github.com/google/syzkaller/blob/master/executor/executor.cc#L685
but we somehow get away with it on linux to date.
When I run it on FreeBSD I got a bunch of flakes:
```
        run.go:63: align0  C                             : OK
        run.go:63: align0 /repeat C                      : OK
        run.go:63: align0 /thr C                         : OK
        run.go:63: align0 /thr/repeat C                  : OK
        run.go:63: align0 none                           : OK
        run.go:63: align0 none C                         : OK
        run.go:63: align0 none/repeat                    : OK
        run.go:63: align0 none/repeat C                  : OK
        run.go:63: align0 none/thr                       : OK
        run.go:63: align0 none/thr C                     : OK
        run.go:63: align0 none/thr/repeat                : FAIL: run 1: call 1 is blocked
        run.go:63: align0 none/thr/repeat C              : OK
        run.go:63: bf  C                                 : OK
        run.go:63: bf /repeat C                          : OK
        run.go:63: bf /thr C                             : OK
        run.go:63: bf /thr/repeat C                      : OK
        run.go:63: bf none                               : OK
        run.go:63: bf none C                             : OK
        run.go:63: bf none/repeat                        : OK
        run.go:63: bf none/repeat C                      : OK
        run.go:63: bf none/thr                           : OK
        run.go:63: bf none/thr C                         : OK
        run.go:63: bf none/thr/repeat                    : FAIL: run 2: call 0 is blocked
        run.go:63: bf none/thr/repeat C                  : OK
        run.go:63: errno  C                              : OK
        run.go:63: errno /repeat C                       : OK
        run.go:63: errno /thr C                          : OK
        run.go:63: errno /thr/repeat C                   : OK
        run.go:63: errno none                            : OK
        run.go:63: errno none C                          : OK
        run.go:63: errno none/repeat                     : OK
        run.go:63: errno none/repeat C                   : OK
        run.go:63: errno none/thr                        : FAIL: run 0: call 3 is blocked
        run.go:63: errno none/thr C                      : OK
        run.go:63: errno none/thr/repeat                 : OK
        run.go:63: errno none/thr/repeat C               : OK
        run.go:63: exit0  C                              : OK
```

We may try to fix this by increasing the timeout. But increasing timeout decreases overall throughput (we wait for longer on every blocked syscall).

Reply:
> I suspect the change to add -B/usr/lib32 is a bug in our gcc port; it shouldn't be necessary when using the base system's copy of clang. Seems fine for the time being though.

I found this complaint and solution on internet, so I guess other users hit it too.
We should try using clang for freebsd.

Reply:
I see. I'll try and think about this problem as I become more familiar with syzkaller internals.